notebooks/4_visualize_forecast.ipynb

```diff
@@ -0,0 +1,1025 @@
+    "\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import geopandas\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "from mpl_toolkits.axes_grid1 import make_axes_locatable\n",
+    "import math"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Navigate to main repository."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Navigate one level up to the main repository\n",
+    "os.chdir(\"..\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Read in path variables from .env."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read environmental variables\n",
+    "env_file = os.path.join(\".env\")\n",
+    "dotenv.load_dotenv(env_file)\n",
+    "\n",
+    "input_dir = os.getenv(\"INPUT_PATH\")\n",
+    "out_dir = os.getenv(\"OUTPUT_PATH\")\n",
+    "countries_path = os.getenv(\"COUNTRIES_PATH\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Read in parameters from config.json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create the path to the forecasted model outputs\n",
+    "with open(\"config.json\") as json_file:\n",
+    "    config = json.load(json_file)\n",
+    "\n",
+    "sim_name = config[\"sim_name\"]\n",
+    "run_name = f\"{sim_name}_forecast\"\n",
+    "\n",
+    "results_dir = f\"{out_dir}/{run_name}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create directories for additional summary statistics and visuals"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results_dir = (\n",
+    "    r\"Q:\\Shared drives\\Pandemic Data\\slf_model\\outputs\\slf_ensemble_calibrated\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not os.path.exists(f\"{out_dir}/summary_stats/{run_name}/\"):\n",
+    "    os.makedirs(f\"{out_dir}/summary_stats/{run_name}/\")\n",
+    "\n",
+    "if not os.path.exists(f\"{results_dir}/figs/\"):\n",
+    "    os.makedirs(f\"{results_dir}/figs/\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Native countries list\n",
+    "native_countries_list = config[\"native_countries_list\"]\n",
+    "\n",
+    "# Country of interest\n",
+    "coi_ISO3 = config[\"coi\"]\n",
+    "\n",
+    "# Run years\n",
+    "start_year = config[\"start_year\"]\n",
+    "end_year = config[\"stop_year\"]\n",
+    "sim_years = config[\"sim_years\"]\n",
+    "num_runs = config[\"run_count\"]"
```

Comment:
This should be the overall number of model runs (num samples).

num_runs = len(glob.glob(f"{results_dir}/*/*/origin_destination.csv"))